### PR TITLE
Add validations to entities and enable admission config PATCH endpoint

### PR DESCRIPTION
**Pull request by rafaelpinabew**

### CODE REVIEW

File: src/controller/AdmissionConfigController.ts [hunk 1]
```diff
@@ -1,53 +1,55 @@
...
     try {
       const { id } = req.params;
       const config = await this.configService.getConfig(id);
+      if (!config) {
+        return next(new NotFoundError('Config not found'));
+      }
       res.status(200).send({ config });
     } catch (error) {
-      res.status(500).send({ error });
+      next(new InternalServerError(error.message));
```
Comment (lutzk) [inline src/controller/AdmissionConfigController.ts:23]: why is that?
why pass error to next mw | requestHandler ?

File: src/business/__tests__/RequestedDocumentService.test.ts [hunk 2]
```diff
@@ -9,29 +9,36 @@ const defaultRequestedDocument = {
...
+      .mockReturnValueOnce({
+        findOne: jest.fn(() => Promise.resolve(defaultRequestedDocument)),
+        // save: jest.fn((updateData) => Promise.resolve({ ...defaultRequestedDocument, ...updateData } as unknown as typeorm.UpdateResult)),
+      })
+  );
 });
 
 describe('Requested Document Service', () => {
   it('should get requested document by id', async () => {
+    jest.spyOn(typeorm, 'getCustomRepository').mockReturnValue({
```
Comment (rafaelpinabew) [inline src/business/__tests__/RequestedDocumentService.test.ts:28]: are we using getCustomRepository method? I thought we are not using it on admissionService? @cemusta

File: src/api/middleware/HttpException.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+export class BadRequestError extends HttpException {
+  status: number;
+
+  message: string;
+
+  constructor(message: string) {
+    super(400, message);
+    this.status = 404;
```
Comment (gosia-cieslewicz) [inline src/api/middleware/HttpException.ts:32]: that should be 400 I believe?

Comment (lutzk) [reply src/api/middleware/HttpException.ts:32]: looks like 
good catch 👍

File: src/controller/AdmissionConfigController.ts (hunk 1 above)
Comment (cemusta) [reply src/controller/AdmissionConfigController.ts:23]: https://expressjs.com/en/guide/error-handling.html

throw new error is ok for normal functions, express catches them

but if its async, it needed to be passed to next(), error passed to next triggers errorHandler

File: src/business/__tests__/RequestedDocumentService.test.ts (hunk 2 above)
Comment (cemusta) [reply src/business/__tests__/RequestedDocumentService.test.ts:25]: Yes good catch.

File: src/controller/AdmissionConfigController.ts (hunk 1 above)
Comment (lutzk) [reply src/controller/AdmissionConfigController.ts:23]: but isnt that for errors thown while executing?
we want not to throw but to send a res with the error

**Review by lutzk — APPROVED** [review]

**Pull request merged**